tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -763,6 +763,67 @@ class ToEtcExpr(
         val opsig = OperT1(Seq(a, BoolT1), BoolT1) // (a, Bool) => Bool
         mkExRefApp(opsig, Seq(sub, act))
 
+      // ******************************************** Variants **************************************************
+      case OperEx(VariantOper.variant, v @ ValEx(TlaStr(tagName)), valueEx) =>
+        val a = varPool.fresh
+        val b = varPool.fresh
+        // (Str, a) => T1a(a) | b
+        val opsig =
+          OperT1(Seq(StrT1, a), VariantT1(RowT1(b, tagName -> a)))
+        // For some reason, we required the field tag: Str to be present in the value type. We should revisit this.
+        mkExRefApp(opsig, Seq(v, valueEx))
+
+      case ex @ OperEx(VariantOper.variant, tag @ _, _) =>
+        throw new TypingInputException(s"The second argument of Variant must be a string, found: $tag", ex.ID)
```

Comment:
I think this is actually the first argument to `Variant`?

```suggestion
        throw new TypingInputException(s"The first argument of Variant must be a string, found: $tag", ex.ID)
```